public/app/features/dashboard/dashgrid/PanelHeader/PanelHeader.tsx

```diff
@@ -1,175 +1,50 @@
-import { AngularComponent, config } from '@grafana/runtime';
-import { ClickOutsideWrapper, Icon, IconName, Tooltip, stylesFactory } from '@grafana/ui';
-import { selectors } from '@grafana/e2e-selectors';
+import { DataLink, PanelData } from '@grafana/data';
+import { Icon } from '@grafana/ui';
 
 import PanelHeaderCorner from './PanelHeaderCorner';
-import { PanelHeaderMenu } from './PanelHeaderMenu';
 
 import { DashboardModel } from 'app/features/dashboard/state/DashboardModel';
 import { PanelModel } from 'app/features/dashboard/state/PanelModel';
 import { getPanelLinksSupplier } from 'app/features/panel/panellinks/linkSuppliers';
-import { getPanelMenu } from 'app/features/dashboard/utils/getPanelMenu';
-import { updateLocation } from 'app/core/actions';
-import { css } from 'emotion';
+import { PanelHeaderNotices } from './PanelHeaderNotices';
+import { PanelHeaderMenuTrigger } from './PanelHeaderMenuTrigger';
+import { PanelHeaderLoadingIndicator } from './PanelHeaderLoadingIndicator';
+import { PanelHeaderMenuWrapper } from './PanelHeaderMenuWrapper';
+import { selectors } from '@grafana/e2e-selectors';
 
 export interface Props {
   panel: PanelModel;
   dashboard: DashboardModel;
   title?: string;
   description?: string;
-  angularComponent?: AngularComponent | null;
   links?: DataLink[];
   error?: string;
   alertState?: string;
   isViewing: boolean;
   isEditing: boolean;
   data: PanelData;
-  updateLocation: typeof updateLocation;
-}
-
-interface ClickCoordinates {
-  x: number;
-  y: number;
-}
-
-interface State {
-  panelMenuOpen: boolean;
-  menuItems: PanelMenuItem[];
+  renderCounter?: number;
 }
 
-export class PanelHeader extends PureComponent<Props, State> {
-  clickCoordinates: ClickCoordinates = { x: 0, y: 0 };
-
-  state: State = {
-    panelMenuOpen: false,
-    menuItems: [],
-  };
-
-  eventToClickCoordinates = (event: React.MouseEvent<HTMLDivElement>) => {
-    return {
-      x: Math.floor(event.clientX),
-      y: Math.floor(event.clientY),
-    };
-  };
-
-  onMouseDown = (event: React.MouseEvent<HTMLDivElement>) => {
-    this.clickCoordinates = this.eventToClickCoordinates(event);
-  };
-
-  isClick = (clickCoordinates: ClickCoordinates) => {
-    return clickCoordinates.x === this.clickCoordinates.x && clickCoordinates.y === this.clickCoordinates.y;
-  };
-
-  onMenuToggle = (event: React.MouseEvent<HTMLDivElement>) => {
-    if (!this.isClick(this.eventToClickCoordinates(event))) {
-      return;
-    }
-
-    event.stopPropagation();
-
-    const { dashboard, panel, angularComponent } = this.props;
-    const menuItems = getPanelMenu(dashboard, panel, angularComponent);
-
-    this.setState({
-      panelMenuOpen: !this.state.panelMenuOpen,
-      menuItems,
-    });
-  };
-
-  closeMenu = () => {
-    this.setState({
-      panelMenuOpen: false,
-    });
-  };
-
+export class PanelHeader extends PureComponent<Props> {
```

Comment:
Would it make sense to just make it function component?

Reply:
That makes a lot of sense.

Reply:
Fixed.